src/main/java/dan/rojas/epam/db/social/db/repository/BatchRepository.java

```diff
@@ -0,0 +1,59 @@
+package dan.rojas.epam.db.social.db.repository;
+
+import dan.rojas.epam.db.social.config.DbConfig;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.ListUtils;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.jdbc.core.BatchPreparedStatementSetter;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Function;
+
+@Slf4j
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@Repository
+@RequiredArgsConstructor
+public class BatchRepository {
+
+  private ExecutorService executor;
+
+  private final JdbcTemplate jdbcTemplate;
+  private final DbConfig dbConfig;
+
+  @PostConstruct
+  public void init() {
+    executor = Executors.newFixedThreadPool(dbConfig.getThreads());
+  }
+
+  public <T> void insert(final String insertStatement,
+                         final List<T> list,
+                         final Function<List<T>, BatchPreparedStatementSetter> mapperFunction) {
+
+    final List<List<T>> partitions = ListUtils.partition(list, dbConfig.getBatchSize());
```

Comment:
Sometimes we do not need heavy config beans with tones of parameters. All we need is couple of values. So we can inject it directly. Of course if your bean itself is too heavy that would be ugly. But if you will follow single responcibilty principle it could make more sense.
Always tend to minimize amount of passed information to the minimum. That will help to refactor later and avoid bad bugs.